### PR TITLE
8343293: Remove the check for <apphome>/jre/lib/libjava.dylib from the launcher's java_md_macosx.m

### PR DESCRIPTION
**Pull request by jaikiran**
Can I please get a review of this change which cleans up a leftover code from the launcher's java_md_macosx.m file?

In https://bugs.openjdk.org/browse/JDK-8329862 changes were done in the launcher code to remove checks for  `<apphome>/jre/lib/` when determining the JDK installation root. Those changes were done in the Unix and Windows files and the macosx one got left out, likely due to an oversight.

The commit in this PR removes that leftover code. No new tests have been added and existing tier1, tier2, tier3 tests continue to pass.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 2 [Reviewers](https://openjdk.org/bylaws#reviewer))

### Issue
 * [JDK-8343293](https://bugs.openjdk.org/browse/JDK-8343293): Remove the check for &lt;apphome&gt;/jre/lib/libjava.dylib from the launcher's java_md_macosx.m (**Bug** - P4)

### Reviewers
 * [Brian Burkhalter](https://openjdk.org/census#bpb) (@bplb - **Reviewer**)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21789/head:pull/21789` \
`$ git checkout pull/21789`

Update a local copy of the PR: \
`$ git checkout pull/21789` \
`$ git pull https://git.openjdk.org/jdk.git pull/21789/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21789`

View PR using the GUI difftool: \
`$ git pr show -t 21789`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21789.diff">https://git.openjdk.org/jdk/pull/21789.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21789#issuecomment-2447351988)
</details>

### CODE REVIEW

**Review by bplb — APPROVED** [review]
Looks fine to me, especially given the testing cited, but this is not my area so I bumped the reviewer count to 2.

File: src/java.base/macosx/native/libjli/java_md_macosx.m [hunk 1]
```diff
@@ -429,18 +429,6 @@ static void MacOSXStartup(int argc, char *argv[]) {
                 return JNI_TRUE;
             }
         }
-        /* ensure storage for path + /jre + NULL */
```
Comment (RogerRiggs) [inline src/java.base/macosx/native/libjli/java_md_macosx.m:432]: The corresponding changes in JDK-8329862 included adding a call to JLI_TraceLauncher.
Such as:
`    JLI_TraceLauncher("Attempt to get JDK installation root from launcher executable path\n");
`
Adding a similar trace message would be consistent across platforms.

Comment (jaikiran) [reply src/java.base/macosx/native/libjli/java_md_macosx.m:432]: That's a good point. I've now updated the PR to add the same log message as the one present in unix/windows variant of this code.

**Review by RogerRiggs — APPROVED** [review]
Thanks fro the extra logging.

**Review by bplb — APPROVED** [review]
Looks fine.

**Comment by jaikiran** [comment]
Thank you Roger and Brian for the reviews.

**Pull request closed without merging**